sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/tasks/TaskManagementSteps.java

```diff
@@ -53,5 +76,152 @@ public TaskManagementSteps(WebDriverHelpers webDriverHelpers) {
           webDriverHelpers.fillAndSubmitInWebElement(
               GENERAL_SEARCH_INPUT, EditCaseSteps.aCase.getUuid());
         });
+
+    When(
+        "^I search last created task by API using Contact UUID and wait for (\\d+) results to be displayed$",
+        (Integer displayedResults) -> {
+          webDriverHelpers.waitUntilElementIsVisibleAndClickable(GENERAL_SEARCH_INPUT);
+          webDriverHelpers.fillAndSubmitInWebElement(
+              GENERAL_SEARCH_INPUT, apiState.getCreatedContact().getUuid());
+          webDriverHelpers.waitUntilNumberOfElementsIsReduceToGiven(TABLE_ROW, displayedResults);
+        });
+
+    When(
+        "^I am checking if all the fields are correctly displayed in the Task Management table$",
+        () -> {
+          org.sormas.e2etests.pojo.api.Task expectedTask = apiState.getCreatedTask();
+          Task actualTask =
+              taskTableRows.stream()
+                  .filter(not(task -> task.getCommentsOnExecution().isEmpty()))
+                  .findFirst()
+                  .orElseThrow();
+          softly
+              .assertThat(apiState.getCreatedContact().getUuid())
+              .containsIgnoringCase(
+                  getPartialUuidFromAssociatedLink(actualTask.getAssociatedLink()));
+          softly
+              .assertThat(actualTask.getAssociatedLink())
+              .containsIgnoringCase(apiState.getCreatedContact().getPerson().getFirstName());
+          softly
+              .assertThat(actualTask.getAssociatedLink())
+              .containsIgnoringCase(apiState.getCreatedContact().getPerson().getLastName());
+          softly
+              .assertThat(actualTask.getTaskContext())
+              .isEqualToIgnoringCase(expectedTask.getTaskContext());
+          softly
+              .assertThat(actualTask.getTaskType())
+              .isEqualTo(properties.getProperty("TaskType." + expectedTask.getTaskType()));
+          softly
```

Comment:
Please consider a way to create an object assert here if possible.

My suggestion would be to update the task object during the test execution. would this be feasible?

Reply:
This is a good point, but here I am comparing two different objects.